test/lib/system_helpers_test.rb

```diff
@@ -60,5 +60,22 @@ module ForemanMaintain
         refute system.check_max_version('katello', '3.18')
       end
     end
+
+    describe 'format_shell_args options' do
+      it 'returns the string without single or double quotes' do
+        config = { 'user' => 'foo', 'password' => 'foopassword1!' }
+        escaped_string = ' user foo password foopassword1\\!'
+        assert_match escaped_string, system.format_shell_args(config)
+      end
+    end
+
+    describe 'shellescape string' do
+      it 'escapes quotes' do
+        password_one = "foo\'bar"
```

Comment:
shouldn't this be 
```suggestion
        password_one = "foo'bar"
```

Reply:
or are you expecting it to escape the `\` too?

Reply:
nope, my mistake, changed this!